hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/CapacityVolumeChoosingPolicy.java

```diff
@@ -58,9 +58,13 @@ public HddsVolume chooseVolume(List<HddsVolume> volumes,
       throw new DiskOutOfSpaceException("No more available volumes");
     }
 
+    List<HddsVolume> volumesWithWriteAllowed =
+        volumes.stream().filter(k -> k.getStorageState() != StorageVolume.VolumeState.READ_ONLY)
+            .collect(Collectors.toList());
+
     AvailableSpaceFilter filter = new AvailableSpaceFilter(maxContainerSize);
 
-    List<HddsVolume> volumesWithEnoughSpace = volumes.stream()
+    List<HddsVolume> volumesWithEnoughSpace = volumesWithWriteAllowed.stream()
         .filter(filter)
         .collect(Collectors.toList());
```

Comment:
Please combine the streams, apply both filters to the same stream.  (Or move the condition to `AvailableSpaceFilter`.)